servicetalk-grpc-protobuf/src/main/java/io/servicetalk/grpc/protobuf/ProtoBufSerializationProviderBuilder.java

```diff
@@ -155,7 +157,7 @@ public GrpcSerializationProvider build() {
         @Override
         public Buffer serialize(final HttpHeaders headers, final T value, final BufferAllocator allocator) {
             addContentHeaders(headers);
-            return serializer.serialize(value, allocator);
+            return serializer.serialize(value, allocator, METADATA_SIZE + value.getSerializedSize());
```

Comment:
If the size is not memoized, `getSerializedSize()` will use reflection to calculate the size. Is that worth the size optimization here?

Reply:
We invoke `getSerializedSize()` anyway to fill in the metadata later. So, we do not pay the reflection cost twice. Without this, profiling shows that we frequently expand the buffer during `writeTo` because it can not fit the serialized proto.